app/src/main/java/org/lineageos/glimpse/viewmodels/MediaViewModel.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-FileCopyrightText: 2023 The LineageOS Project
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.lineageos.glimpse.viewmodels
+
+import androidx.lifecycle.*
+import androidx.lifecycle.ViewModelProvider.AndroidViewModelFactory.Companion.APPLICATION_KEY
+import androidx.lifecycle.viewmodel.initializer
+import androidx.lifecycle.viewmodel.viewModelFactory
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.*
+import org.lineageos.glimpse.GlimpseApplication
+import org.lineageos.glimpse.repository.MediaRepository
+
+class MediaViewModel(
+    private val savedStateHandle: SavedStateHandle, private val mediaRepository: MediaRepository
+) : ViewModel() {
+    private val mediaPositionInternal = savedStateHandle.getLiveData<Int>(MEDIA_POSITION_KEY)
```

Comment:
I'd love to use flows here as well, we can use getInt and get the raw data and set it to a flow, should work :)